File: src/py/q5.py
import math

def lcm(a, b):
    return abs(a * b)// math.gcd(a, b)

def lcmrange(n):
    res = 1
    for i in range(1, n + 1):
        res = lcm(res, i)
    return res

print(lcmrange(20))

File: src/R/q5.R
lcm <- function(a, b) {
  return(abs(a * b) / gcd(a, b))
}

lcm_range <- function(n) {
  res <- 1
  for (i in 1:n) {
    res <- lcm(res, i)
  }
  return(res)
}

print(lcm_range(20))
